### PR TITLE
Add initial Linux support

### PR DESCRIPTION
**Pull request by Babouchot**
Early fixes and additions for the project to compile and run on Linux.

### CODE REVIEW

File: Assets/Scripts/Sharing/DriveAccess.cs [hunk 1]
```diff
@@ -99,6 +99,10 @@ private class ParentNotFound : Exception {
       case RuntimePlatform.OSXEditor:
         Debug.LogError("Host id not implemented for macOS");
         return "macOS-unknown";
+      case RuntimePlatform.LinuxPlayer:
+      case RuntimePlatform.LinuxEditor:
+        Debug.LogError("Host id is not implemented for Linux");
+        return "Linux-unknown";
```
Comment (TimAidley) [inline Assets/Scripts/Sharing/DriveAccess.cs:105]: Note that having a working host id is required for proper working of the Google Drive backup system.

Comment (Babouchot) [reply Assets/Scripts/Sharing/DriveAccess.cs:105]: Sure adding proper host id handling was on my todo list. The main goal here was to have a first version that compiles and that runs without exceptions

**Comment by mikeage** [comment]
You might also find #21 to be relevant; the fixes for adb and rsync should help you as well (and it's possibly that the background build might work on Linux even though it doesn't work on OS X; I don't have an environment to test)

**Comment by Babouchot** [comment]
> You might also find #21 to be relevant; the fixes for adb and rsync should help you as well (and it's possibly that the background build might work on Linux even though it doesn't work on OS X; I don't have an environment to test)

@mikeage I just tested my branch rebased on top of yours and indeed it seems to be working well :ok_hand:

**Comment by mikeskydev** [comment]
Superseded by #32

**Pull request closed without merging**